app/jobs/populate_register_data_in_db_job.rb

```diff
@@ -1,6 +1,30 @@
 class PopulateRegisterDataInDbJob < ApplicationJob
   queue_as :default
 
+
+  module Exceptions
+    class FrontendInvalidRegisterError < StandardError; end
+  end
+
+  module RegisterClientExtensions
+    include Exceptions
+    def refresh_data
+      begin
+        super
+      rescue InvalidRegisterError
+        # If register data is invalid we want to delete existing entries and records to force a full reload
+        register_id = @data_store.instance_variable_get(:@register).id
```

Comment:
I think this is acceptable in tests but perhaps not in production code. Can the `InvalidRegisterException` be caught by wrapping the call to `refreshData` in a try-catch block?

Reply:
No, that is not sufficient because `refresh_data` is also called by the client library itself when `RegisterClient` is initialised in: https://github.com/openregister/registers-ruby-client/blob/2674bb6292d9d6b582c92ac62722bfed8e7a6982/lib/register_client.rb#L12 and this is the first exception thrown.
I agree there should be a better way to access the register ID, since it is  passed to the `perform` method but not available at the class level to be injected here.

Reply:
@gidsg I'm not following your explanation.